superset/db_engine_specs/base.py

```diff
@@ -437,6 +437,14 @@ class BaseEngineSpec:  # pylint: disable=too-many-public-methods
     # Driver-specific exception that should be mapped to OAuth2RedirectError
     oauth2_exception = OAuth2RedirectError
 
+    # Does the query id related to the connection?
+    # The default value is True, which means that the query id is determined when
+    # the connection is created.
+    # When this is changed to false in a DB engine spec it means the query id
+    # is determined only after the specific query is executed and it will update
+    # the `cancel_query` value in the `extra` field of the `query` object
+    is_query_id_associated_connect = True
```

Comment:
I like the comment, however, I find the property name difficult to understand. Could we rename this to something more descriptive, like `has_query_id_before_execute` or something similar.